LoopKit/SettingsStore.swift

```diff
@@ -269,8 +269,8 @@ extension StoredSettings: Codable {
         self.init(date: try container.decode(Date.self, forKey: .date),
                   dosingEnabled: try container.decode(Bool.self, forKey: .dosingEnabled),
                   glucoseTargetRangeSchedule: try container.decodeIfPresent(GlucoseRangeSchedule.self, forKey: .glucoseTargetRangeSchedule),
-                  preMealTargetRange: try container.decodeIfPresent(DoubleRange.self, forKey: .preMealTargetRange)?.quantityRange(for: StoredSettings.codingGlucoseUnit),
-                  workoutTargetRange: try container.decodeIfPresent(DoubleRange.self, forKey: .workoutTargetRange)?.quantityRange(for:  StoredSettings.codingGlucoseUnit),
+                  preMealTargetRange: try container.decodeIfPresent(DoubleRange.self, forKey: .preMealTargetRange)?.quantityRange(for: bloodGlucoseUnit ?? StoredSettings.codingGlucoseUnit),
```

Comment:
I would make `bloodGlucoseUnit` a required property of the encoded/decoded data.

So, a few lines above (line 265-268), change:
```
        var bloodGlucoseUnit: HKUnit?
        if let bloodGlucoseUnitString = try container.decodeIfPresent(String.self, forKey: .bloodGlucoseUnit) {
            bloodGlucoseUnit = HKUnit(from: bloodGlucoseUnitString)
        }
```
to something like:
```
       let bloodGlucoseUnit = HKUnit(from: try container.decode(String.self, forKey: .bloodGlucoseUnit))
```

And, in the `encode` function below, change:
```
        try container.encodeIfPresent(bloodGlucoseUnit?.unitString, forKey: .bloodGlucoseUnit)
```
to something like:
```
        try container.encode((bloodGlucoseUnit ?? StoredSettings.codingGlucoseUnit).unitString, forKey: .bloodGlucoseUnit)
```
Does that make sense?